src/lightcurvelynx/models/sed_curve_model.py

```diff
@@ -0,0 +1,520 @@
+"""Model that generate the SED or bandflux of a source based on given observer frame
+SED curves at given wavelengths.
+
+Note: If you are interested in generating light curves from band-level curves, use
+the LightcurveTemplateModel in src/lightcurvelynx/models/lightcurve_template_model.py
+instead.
+"""
+
+import logging
+import warnings
+from pathlib import Path
+
+import astropy.units as u
+import numpy as np
+from scipy.interpolate import RectBivariateSpline
+
+from lightcurvelynx.astro_utils.unit_utils import flam_to_fnu
+from lightcurvelynx.math_nodes.given_sampler import GivenValueSampler
+from lightcurvelynx.models.physical_model import SEDModel
+
+logger = logging.getLogger(__name__)
+
+
+class LightcurveSEDData:
```

Comment:
Sorry for the naming concern, but let's maybe refer to it without "lightcurve" term, which is usually applied to bandflux/mag data. Maybe TemporalSEDData or just SEDData? The same goes to lightcurves argument, etc